tests/test_agents.py

```diff
@@ -969,6 +970,7 @@ def test_answers_are_striped() -> None:
     assert not response.session.contexts[0].text.text
     assert response.session.contexts[0].text.doc is not None
     assert response.session.contexts[0].text.doc.embedding is None
+    assert response.session.contexts[0].extra_field is not None  # type: ignore[attr-defined]
```

Comment:
Can you add an assertion message here as informal docs, something like `, "Extra fields should be propagated so we can XYZ"`